activerecord/lib/active_record/fixtures.rb

```diff
@@ -986,7 +986,7 @@ def setup_fixtures(config = ActiveRecord::Base)
       # Load fixtures for every test.
       else
         ActiveRecord::FixtureSet.reset_cache
-        @@already_loaded_fixtures[self.class] = nil
+        @@already_loaded_fixtures = {}
         @loaded_fixtures = load_fixtures(config)
```

Comment:
Would make sense to set `@@already_loaded_fixtures` to the `@loaded_fixtures` here, anticipating that the next test will `run_in_transaction?`.